internal/include/jemalloc/internal/jemalloc_internal_decls.h:21:14: fatal error: 'os/lock.h' file not found

I pulled master of c-jemalloc, on OSX 10.11.6, but it doesn't seem to build...

~~~
jaten@jatens-MacBook-Pro ~/go/src/github.com/cockroachdb/c-jemalloc (master) $ go install
go install
# github.com/cockroachdb/c-jemalloc
In file included from ./arena.c:2:
In file included from darwin_includes/internal/include/jemalloc/internal/jemalloc_internal.h:5:
internal/include/jemalloc/internal/jemalloc_internal_decls.h:21:14: fatal error: 'os/lock.h' file not found
jaten@jatens-MacBook-Pro ~/go/src/github.com/cockroachdb/c-jemalloc (master) $ 
~~~

the compile trace in the first comment of https://github.com/harbour/core/issues/128  makes me think that os/lock.h may have been introduced in OSX 10.12.

`../../../../../include/hbatomic.h:477:10: warning: 'OSSpinLock' is deprecated: first deprecated in macOS 10.12 - Use os_unfair_lock() from <os/lock.h>`

--- a/darwin_includes/internal/include/jemalloc/internal/jemalloc_internal_defs.h
+++ b/darwin_includes/internal/include/jemalloc/internal/jemalloc_internal_defs.h
@@ -59,7 +59,7 @@
 /*
  * Defined if os_unfair_lock_*() functions are available, as provided by Darwin.
  */
-#define JEMALLOC_OS_UNFAIR_LOCK 
+/* #undef JEMALLOC_OS_UNFAIR_LOCK */
 
 /*
  * Defined if OSSpin*() functions are available, as provided by Darwin, and
@@ -68,7 +68,7 @@
 #define JEMALLOC_OSSPIN 
 
 /* Defined if syscall(2) is usable. */
-/* #undef JEMALLOC_USE_SYSCALL */
+#define JEMALLOC_USE_SYSCALL 
 
 /*
  * Defined if secure_getenv(3) is available.

--- a/import.sh
+++ b/import.sh
@@ -9,7 +9,7 @@ curl -sL https://github.com/jemalloc/jemalloc/releases/download/4.4.0/jemalloc-4
 
 # You need to manually run the following code.
 # on OSX:
-# (cd internal && ./configure --enable-prof --with-jemalloc-prefix="")
+# (cd internal && MACOSX_DEPLOYMENT_TARGET=10.9 ./configure --enable-prof --with-jemalloc-prefix='')
 # <compare "Build parameters" in internal/Makefile to cgo flags in cgo_flags.go> and adjust the latter.
 # rm -r darwin_includes
 # git clean -Xn -- internal/include/jemalloc | sed 's/.* //' | xargs -I % rsync -R % darwin_includes/

